flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/FunctionGenerator.scala

```diff
@@ -854,7 +854,7 @@ class FunctionGenerator private(config: TableConfig) {
     operandTypes: Seq[LogicalType],
     resultType: LogicalType)
   : Option[CallGenerator] = sqlOperator match {
-    // built-in scalar function
+    case JSON_VALUE => Some(new JsonValueCallGen())
```

Comment:
This seems to be the easiest answer to the problem of not multiplying out a lot of argument type combinations when you don't care about the specific argument types.

Reply:
this is hard to find, wouldn't be the layer above be more appropriate where CAST and other custom one are? or would that be more than a single line change then?

Reply:
No, that's also fine. Will do.